content/automate/ManageIQ/System/Event/EmsEvent/RHEVM.class/__methods__/update_migration_status.rb

```diff
@@ -0,0 +1,34 @@
+module ManageIQ
+  module Automate
+    module System
+      module Event
+        module EmsEvent
+          module RHEVM
+            class UpdateMigrationStatus
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                update_migration_status
+              end
+
+              private
+
+              def update_migration_status
+                if @handle.root['event_type'] == 'VM_MIGRATION_FAILED_FROM_TO'
+                  @handle.root['ae_result'] = 'error'
```

Comment:
@tinaafitz This line do not mark migration request as failed. Can you help me to understand how to fail it when `VM_MIGRATION_FAILED_FROM_TO` arrives from the provider?

Reply:
@pkliczewski The issue here is that the VM Migration state machine and event processing are entirely separate.  

The vm migrate state machine is should fail on the execute:  
https://github.com/tinaafitz/manageiq/blob/master/app/models/vm_migrate_task.rb#L65

Instead of marking the status as migrated:
https://github.com/tinaafitz/manageiq/blob/master/app/models/vm_migrate_task.rb#L72

Reply:
@pkliczewski Do you have an environment setup where I can reproduce the migration failure?

Reply:
It seems that we need to do it as part of state machine. Do you have any suggestions or examples how to?

Reply:
@pkliczewski you can mimic 
https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/System/Event/EmsEvent/RHEVM.class/__methods__/update_vm_import_status.rb#L42 where you can set a customer attribute by assigning the migration result to a unique key. Don't fail the request here regardless the migration result.

Then you can mimic https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/Infrastructure/VM/Transform/StateMachines/VmImport.class/__methods__/wait_for_vm_import.rb#L19 where they check the `import_status` to decide to retry, complete, or fail the `wait_for_vim_import` step. You should do the same in https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/Infrastructure/VM/Migrate/StateMachines/Methods.class/__methods__/checkmigration.rb#L27

Lastly, as @tinaafitz mentioned earlier, in https://github.com/ManageIQ/manageiq/blob/master/app/models/vm_migrate_task.rb#L72 change the state from `migrated` to `migrating`. 

Let's see how it works. Later we can work together to find out how it can apply to VMware too.